### PR TITLE
feat: implement ls-refs upload-pack command

### PR DESCRIPTION
**Pull request by g4s8**
Implement `ls-refs` command handling of `git-upload-pack` service:
 * implement command parsing of git client.
 * add `LsRefsSlice` to handle `ls-refs` command of `git-upload-pack`
service.
 * add `TmpResource` to copy git repository from storage to temp
   directory.
 * implement command routing for upload-pack service.
 * enable and fix integration test for `ls-refs` command.

### CODE REVIEW

File: src/main/java/com/artipie/git/LsRefsSlice.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        final Publisher<ByteBuffer> body) {
+        final Path tmp;
+        try {
+            tmp = Files.createTempDirectory(LsRefsSlice.class.getName());
+        } catch (final IOException iex) {
+            throw new ArtipieIOException(iex);
+        }
+        return new AsyncResponse(
+            new TmpResource(tmp).<Response>with(
+                tsto -> new Copy(this.storage).copy(tsto).thenApply(
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/git/LsRefsSlice.java:62]: could you please explain why do we need to copy all the repository into temp location?

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@g4s8 one question

File: src/main/java/com/artipie/git/LsRefsSlice.java (hunk 1 above)
Comment (g4s8) [reply src/main/java/com/artipie/git/LsRefsSlice.java:62]: jGit library supports two kinds of repository storage by default: file-based repository and DFS repository. `FileRepository` is a simple solution now, later we can implement `Repository` interface for `Storage`

Comment (olenagerasimova) [reply src/main/java/com/artipie/git/LsRefsSlice.java:62]: @g4s8 thanks, got it. Please, fix the build and we are good to merge

**Comment by g4s8** [comment]
@olenagerasimova please check all tests passes

**Review by olenagerasimova — APPROVED** [review]
@g4s8 thanks

**Pull request merged**